src/main/java/io/apicurio/datamodels/core/validation/rules/other/SecurityRequirementUniquenessValidationRule.java

```diff
@@ -0,0 +1,134 @@
+            // context).
+            for (int i = 0; i < currentSecurityRequirementNames.size(); i++) {
+                String secReqName = currentSecurityRequirement.getSecurityRequirementNames().get(i);
+                
+                for (int j = 0; j < otherSecurityRequirementNames.size(); j++) {
+                    String otherSecReqName = otherSecurityRequirement.getSecurityRequirementNames().get(j);
+                    if (secReqName.equals(otherSecReqName)) { // if they have the same name eg: [pet_auth] equals [pet_auth]
+                        // now get the scopes for each one of these sr names eg: ["read:pets", "write:pets"] ["write:pets","read:pets"]
+                        List<String> secReqScopes = currentSecurityRequirement.getScopes(secReqName);
+                        List<String> otherSecReqScopes = otherSecurityRequirement.getScopes(otherSecReqName);
+                        // sort them
+                        Collections.sort(secReqScopes);
+                        Collections.sort(otherSecReqScopes);
+                        if (secReqScopes.equals(otherSecReqScopes)) { // compare them
+                            matchesFound += 1; // increment this variable when the scopes match
+                        }
+                    }
+                }
+            }
+            // if the number of matches found == the number of sec requirements for the current one
+            if (matchesFound == currentSecurityRequirementNames.size()) {
+                // Then we have a duplicated security requirement
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param currentSecurityRequirement
+     * @param allRequirements
+     * @return true if duplicates are found
+     */
+    private boolean checkForDuplicates(SecurityRequirement currentSecurityRequirement,
+            List<SecurityRequirement> allRequirements) {
+        ArrayList<SecurityRequirement> clonedList = new ArrayList<SecurityRequirement>();
+        clonedList.addAll(allRequirements);
+        clonedList.remove(currentSecurityRequirement); // remove the current one to examine siblings
+        for (int i = 0; i < clonedList.size(); i++) {
+            // get the SecurityRequirement to compare against
+            SecurityRequirement otherSecurityRequirement = clonedList.get(i);
+            // do a compare to check if the two are equal
+               if(isEqualTo(currentSecurityRequirement, otherSecurityRequirement)) {
+                   return true;
+               }
+        }
+        return false;
+    }
+
+    /**
+     * @param node
+     * @see io.apicurio.datamodels.combined.visitors.CombinedAllNodeVisitor#visitSecurityRequirement(io.apicurio.datamodels.core.models.common.SecurityRequirement)
+     */
+    public void visitSecurityRequirement(SecurityRequirement node) {
+        // may also be a root document
```

Comment:
Might be worth introducing a new model interface for this - something like `ISecurityRequirementParent` with a method e.g. `public List<SecurityRequirement> getSecurity()` - then the root `Document` and `Operation` models can implement it.

Reply:
Ive addresses other feedback. Will look at this next once I put some tests in

Reply:
@EricWittmann There is already a model interface for this https://github.com/Apicurio/apicurio-data-models/blob/master/src/main/java/io/apicurio/datamodels/core/models/common/ISecurityRequirementParent.java  and with the method `getSecurityRequirements()`.